packages/gatsby/src/services/graphql-typegen.ts

```diff
@@ -0,0 +1,62 @@
+import reporter from "gatsby-cli/lib/reporter"
+import { EventObject } from "xstate"
+import { IBuildContext } from "../internal"
+import { IDataLayerContext, IQueryRunningContext } from "../state-machines"
+import {
+  writeGraphQLFragments,
+  writeGraphQLSchema,
+} from "../utils/graphql-typegen/file-writes"
+import { writeTypeScriptTypes } from "../utils/graphql-typegen/ts-codegen"
+
+export async function graphQLTypegen(
+  {
+    program,
+    store,
+    parentSpan,
```

Comment:
I would also include reporter here instead of importing it.
```suggestion
    program,
    store,
    reporter,
    parentSpan,
```